### PR TITLE
[path_provider] support the v2 embedder

### PR DESCRIPTION
**Pull request by audkar**
## Description

Migrate android path_provider plugin to the new embedding method.

## Related Issues

flutter/flutter#41844

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by audkar** [comment]
Currently existing drive test fails because of exception (before my changes): 
```
Expected: throws <Instance of 'UnsupportedError'>
    Actual: <Instance of 'Future<Directory>'>
    Which: threw MissingPluginException:<MissingPluginException(No implementation 
                found for method getLibraryDirectory on channel plugins.flutter.io/path_provider)>
```

this [test](https://github.com/flutter/plugins/blob/master/packages/path_provider/example/test_driver/path_provider.dart#L59) expects `UnsupportedError`. Problem is that thrown error is different. `UnsupportedError` vs `MissingPluginException`. Is this changed how platform is dealing with [result.notImplemented()](https://github.com/flutter/plugins/blob/master/packages/path_provider/android/src/main/java/io/flutter/plugins/pathprovider/PathProviderPlugin.java#L45)?

**Comment by audkar** [comment]
Also `drive` tests are currently executed only for old Plugin implementation. Is there any way to make them run on a new plugin also? (I tested them manually by switching `MainActivity` in AndroidManifest) but what permanent solution could be used to test both plugin implementations with drive test?

**Comment by collinjackson** [comment]
You can include an EmbeddingV1ActivityTest.java for the old embedder and MainActivityTest.java for the new embedder and CI will run both of them. See https://github.com/flutter/plugins/pull/2160 for an example.

**Comment by audkar** [comment]
@collinjackson when I add `e2e` tests I get:
```
/home/audrius/.pub-cache/hosted/pub.dartlang.org/e2e-0.2.1/android/src/main/java/dev/flutter/plugins/e2e/E2EPlugin.java:35: error: cannot access LifecycleOwner
        binding.getApplicationContext(), binding.getFlutterEngine().getDartExecutor());
               ^
  class file for android.arch.lifecycle.LifecycleOwner not found
```
Is current e2e version compatible with stable channel and `Lifecycle` dependency problem?

File: packages/path_provider/android/build.gradle [hunk 1]
```diff
@@ -45,3 +45,28 @@ android {
         disable 'InvalidPackage'
     }
 }
+
+// TODO(amirh): Remove this hack once androidx.lifecycle is included on stable. https://github.com/flutter/flutter/issues/42348
+afterEvaluate {
```
Comment (xster) [inline packages/path_provider/android/build.gradle]: with this temporary tweak, the error you were pointing at should be gone on stable. Would you like to sync your PR up to master and try the tests again?

**Comment by audkar** [comment]
Currently I have no time to work on this PR. Feel free to close or modify it.

**Comment by cyanglaz** [comment]
v2 embedding has been supported in path provider from https://github.com/flutter/plugins/pull/2284. I'm closing this PR

**Pull request closed without merging**